cpp/cmake/thirdparty/get_arrow.cmake

```diff
@@ -192,8 +192,23 @@ function(find_and_configure_arrow VERSION BUILD_STATIC ENABLE_S3 ENABLE_ORC ENAB
           if (TARGET cudf::arrow_static AND (NOT TARGET arrow_static))
               add_library(arrow_static ALIAS cudf::arrow_static)
           endif()
+          if (NOT TARGET arrow::flatbuffers)
+            add_library(arrow::flatbuffers INTERFACE IMPORTED)
+          endif()
+          if (NOT TARGET arrow::hadoop)
+            add_library(arrow::hadoop INTERFACE IMPORTED)
+          endif()
         ]=]
     )
+    if(NOT TARGET xsimd)
```

Comment:
Would this part be different if we used `BUNDLED` instead of `AUTO` for xsimd on line 124?

https://github.com/rapidsai/cudf/blob/9223191c12d1d4cf9532310a223a09f540b54c3e/cpp/cmake/thirdparty/get_arrow.cmake#L124

I recommended using `AUTO` in #11513 because it seemed like a more general / safer option. After further inspection I realized that most of Arrow's build scripts were using `BUNDLED`.

Reply:
No in either case we would need this.